### DISCUSSION TITLE
Default casing for permissions

### DISCUSSION THREAD

**Discussion by olivernybroe** [spatie/laravel-permission#1981] [General]
Currently the documentation uses spaces for permissions, however I personally don't like the syntax of that and especially when used in a middleware

```php
Route::group(['middleware' => ['can:publish articles']], function () {
    //
});
```

Personally I think kebab-case is better

```php
Route::group(['middleware' => ['can:publish-articles']], function () {
    //
});
```

The reason for taking this discussion up, is to change what the default casing is. Another solution could be to make it casing agnostic, e.g. I can write `publish articles`, `publish-articles`, `publish_articles`, `publish_articles`, `publishArticles` and all would refer to the same permission 🤔

**Comment by erikn69** [comment]
There is no rule, it is open for everyone to use as they see fit, changing that maybe would be a breaking change for many(i'm using snake_case)

**Reply by olivernybroe** [reply]
Yep, however people typically tries to follow what the docs do and see that as a standard. 

Yes, my other proposal for allowing all the cases would be a BC.
